src/common/address.py

```diff
@@ -50,9 +50,15 @@ def from_relative(
             go_up = address.count("~")
             path = relative_path.copy()
             while go_up != 0:
-                path.pop(0)
+                if len(path) == 0:
+                    raise ApplicationException("Trying to go outside document.")
```

Comment:
This message could be more specific?
For example; 
"Invalid relative reference. Traversing outside a contained document with '~' is not supported"